### PR TITLE
#102 Rectangular selection

### LINKED ISSUE #102: Marking support on bars
The experience should be the same as the Spotfire bar chart.

If no color by or single color:
- User must be able to mark a bar by clicking on a bar
- User must be able to mark multiple bars by clicking on bars while holding a key
- User must be able to mark multiple bars by drawing a rectangle selection
- User must be able to unmark a bar by clicking outside bars

If color by axis set:
- User must be able to mark a bar segment by clicking on a segment
- User must be able to mark multiple bar segments by click on segments while holding a key
- User must be able to mark multiple segments by drawing a rectangle selection
- User must be able to unmark segments by clicking outside bars

See [Spotfire document about marking](https://docs.tibco.com/pub/sfire-analyst/latest/doc/html/en-US/TIB_sfire-analyst_UsersGuide/index.htm#t=vis%2Fvis_marking_in_visualizations.htm)

### PR DESCRIPTION
**Pull request by MajaLinder**
closes #102

This PR includes implementation of:
- Drawing a rectangle on the chart
- Marking the bar segments contained inside the rectangle when releasing the mouse button
- Unmarking bars when clicking in the background, outside of the bars

image

### CODE REVIEW

File: src/stackedBars.ts [hunk 1]
```diff
@@ -77,8 +79,16 @@ export function renderStackedBars(pareto: Pareto, settings: Settings) {
             .attr("width", bBox.width + 2 * padding)
             .attr("stroke", "#000")
             .attr("stroke-width", settings.style.onMouseOverBox.strokeWidth)
+            .attr("shape-rendering", "crispEdges")
             .attr("fill", "none");
     }
+
+    rectangularSelection({
+        clearMarking: settings.clearMarking,
```
Comment (AstridBer) [inline src/stackedBars.ts:87]: I get this error here, is it the same for you? 

image

Comment (MajaLinder) [reply src/stackedBars.ts:87]: Did you pull the latest code? It should've been fixed in the last commit :)

Comment (AstridBer) [reply src/stackedBars.ts:87]: Seems like I didn't, works now thanks!

**Review by AstridBer — APPROVED** [review]
Nice work!

**Pull request merged**